### ISSUE TITLE
Non-local address on neighbor generate non-related alerts 

### ISSUE THREAD

**Issue by paunadeu** [Exa-Networks/exabgp#974]
##### Summary
Trying to configure neighbor with a non local-address configured gives a error that can lead to confusion.

##### OS
```
MacOS 10.15.4 (Catalina)
(venv)  paunadeurabat@MacBook-Pro-de-Pau  ~/PycharmProjects/RIN   master  uname -a
Darwin MacBook-Pro-de-Pau.local 19.4.0 Darwin Kernel Version 19.4.0: Wed Mar  4 22:28:40 PST 2020; root:xnu-6153.101.6~15/RELEASE_X86_64 x86_64
```

##### ExaBGP Version
```
(venv)  paunadeurabat@MacBook-Pro-de-Pau  ~/PycharmProjects/RIN   master  exabgp --version   
ExaBGP : 4.2.6
Python : 3.8.1 (v3.8.1:1b293b6006, Dec 18 2019, 14:08:53)  [Clang 6.0 (clang-600.0.57)]
Uname  : Darwin MacBook-Pro-de-Pau.local 19.4.0 Darwin Kernel Version 19.4.0: Wed Mar  4 22:28:40 PST 2020; root:xnu-6153.101.6~15/RELEASE_X86_64 x86_64
Root   : /Users/paunadeurabat/PycharmProjects/RIN/venv
>
```

##### Installation
```
pip with virtualenv
Using: pip3 install exabgp
```

##### Environment 
```
empty
```

##### Configuration
```
neighbor 213.242.114.114 {
    router-id 213.242.114.113;
    local-address 213.242.114.113;
    local-as 3356;
    peer-as 33932;
}
```

##### Program output
```
on to 213.242.114.114:179 failed
23:59:50 | 13653  | outgoing-1    | Could not bind to local ip 213.242.114.113 - [Errno 49] Can't assign requested address
argument must be an int, or have a fileno() method.

Traceback (most recent call last):
  File "/Users/paunadeurabat/PycharmProjects/RIN/venv/lib/python3.8/site-packages/exabgp/application/control.py", line 245, in run
    self.loop()
  File "/Users/paunadeurabat/PycharmProjects/RIN/venv/lib/python3.8/site-packages/exabgp/application/control.py", line 218, in loop
    ready = self.read_on(reading)
  File "/Users/paunadeurabat/PycharmProjects/RIN/venv/lib/python3.8/site-packages/exabgp/application/control.py", line 104, in read_on
    poller.register(io, select.POLLIN | select.POLLPRI | select.POLLHUP | select.POLLNVAL | select.POLLERR)
TypeError: argument must be an int, or have a fileno() method.
23:59:51 | 13653  | process       | process api-internal-cli-67758e94 ended, restarting it
23:59:51 | 13653  | process       | terminating process api-internal-cli-67758e94
23:59:51 | 13653  | process       | forked process api-internal-cli-67758e94
argument must be an int, or have a fileno() method.

Traceback (most recent call last):
  File "/Users/paunadeurabat/PycharmProjects/RIN/venv/lib/python3.8/site-packages/exabgp/application/control.py", line 245, in run
    self.loop()
  File "/Users/paunadeurabat/PycharmProjects/RIN/venv/lib/python3.8/site-packages/exabgp/application/control.py", line 218, in loop
    ready = self.read_on(reading)
  File "/Users/paunadeurabat/PycharmProjects/RIN/venv/lib/python3.8/site-packages/exabgp/application/control.py", line 104, in read_on
    poller.register(io, select.POLLIN | select.POLLPRI | select.POLLHUP | select.POLLNVAL | select.POLLERR)
TypeError: argument must be an int, or have a fileno() method.
23:59:51 | 13653  | process       | process api-internal-cli-67758e94 ended, restarting it
23:59:51 | 13653  | process       | terminating process api-internal-cli-67758e94
23:59:51 | 13653  | process       | forked process api-internal-cli-67758e94
23:59:51 | 13653  | process       | Too many death for api-internal-cli-67758e94 (5) terminating program
23:59:51 | 13653  | reactor       | Problem when sending message(s) to helper program, stopping
23:59:51 | 13653  | reactor       | performing shutdown
23:59:51 | 13653  | process       | terminating process api-internal-cli-67758e94
```

##### Steps to reproduce
Maybe more accurate error message should be displayed to avoid user confusion:
```
23:59:50 | 13653  | outgoing-1    | Could not bind to local ip 213.242.114.113 - [Errno 49] Can't assign requested address
```
I think this message should be displayed also without -d, therefore messages (POLLIN, POLLHUP, etc...) causes confusion

**Comment by paunadeu** [comment]
```
(venv)  paunadeurabat@MacBook-Pro-de-Pau  ~/PycharmProjects/RIN   master  exabgp ./exabgp.conf -d
00:05:41 | 13840  | welcome       | Thank you for using ExaBGP
00:05:41 | 13840  | version       | 4.2.6   
00:05:41 | 13840  | interpreter   | 3.8.1 (v3.8.1:1b293b6006, Dec 18 2019, 14:08:53)  [Clang 6.0 (clang-600.0.57)]
00:05:41 | 13840  | os            | Darwin MacBook-Pro-de-Pau.local 19.4.0 Darwin Kernel Version 19.4.0: Wed Mar  4 22:28:40 PST 2020; root:xnu-6153.101.6~15/RELEASE_X86_64 x86_64
00:05:41 | 13840  | installation  | /Users/paunadeurabat/PycharmProjects/RIN/venv
00:05:41 | 13840  | cli control   | named pipes for the cli are:
00:05:41 | 13840  | cli control   | to send commands  /var/run/exabgp.in
00:05:41 | 13840  | cli control   | to read responses /var/run/exabgp.out
00:05:41 | 13840  | configuration | performing reload of exabgp 4.2.6
00:05:41 | 13840  | configuration | > neighbor         | '213.242.114.114'
00:05:41 | 13840  | configuration | . router-id        | '213.242.114.113'
00:05:41 | 13840  | configuration | . local-address    | '192.168.88.254'
00:05:41 | 13840  | configuration | . local-as         | '3356'
00:05:41 | 13840  | configuration | . peer-as          | '33932'
00:05:41 | 13840  | configuration | < neighbor         | 
00:05:41 | 13840  | reactor       | new peer: neighbor 213.242.114.114 local-ip 192.168.88.254 local-as 3356 peer-as 33932 router-id 213.242.114.113 family-allowed in-open
00:05:41 | 13840  | reactor       | loaded new configuration successfully
00:05:41 | 13840  | process       | forked process api-internal-cli-392fd5ca
00:05:41 | 13840  | reactor       | initialising connection to peer-1
00:05:41 | 13840  | outgoing-1    | attempting connection to 213.242.114.114:179
00:05:41 | 13840  | outgoing-1    | sending TCP payload ( 177) FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF 00B1 0104 0D1C 00B4 D5F2 7271 9402 0601 0400 0100 0102 0601 0400 0100 0202 0601 0400 0100 0402 0601 0400 0100 8002 0601 0400 0100 8402 0601 0400 0100 8502 0601 0400 0100 8602 0601 0400 0200 0102 0601 0400 0200 0202 0601 0400 0200 0402 0601 0400 0200 8002 0601 0400 0200 8502 0601 0400 0200 8602 0601 0400 1900 4102 0601 0400 1900 4602 0601 0440 0400 4702 0601 0440 0400 4802 0641 0400 000D 1C02 0206 00
00:05:41 | 13840  | outgoing-1    | >> OPEN version=4 asn=3356 hold_time=180 router_id=213.242.114.113 capabilities=[Multiprotocol(ipv4 unicast,ipv4 multicast,ipv4 nlri-mpls,ipv4 mpls-vpn,ipv4 rtc,ipv4 flow,ipv4 flow-vpn,ipv6 unicast,ipv6 multicast,ipv6 nlri-mpls,ipv6 mpls-vpn,ipv6 flow,ipv6 flow-vpn,l2vpn vpls,l2vpn evpn,bgp-ls bgp-ls,bgp-ls bgp-ls-vpn), Extended Message(65535), ASN4(3356)]
00:05:41 | 13840  | ka-outgoing-1 | receive-timer 60 second(s) left
00:05:41 | 13840  | outgoing-1    | outgoing-1 192.168.88.254-213.242.114.114, closing connection
00:05:41 | 13840  | outgoing-1    | outgoing-1 192.168.88.254-213.242.114.114 213.242.114.114 lost TCP session with peer
00:05:41 | 13840  | outgoing-1    | peer reset, message [closing connection] error[the TCP connection was closed by the remote end]
00:05:41 | 13840  | outgoing-1    | outgoing-1 192.168.88.254-213.242.114.114, closing connection
00:05:41 | 13840  | reactor       | initialising connection to peer-1
00:05:41 | 13840  | outgoing-2    | attempting connection to 213.242.114.114:179
00:05:41 | 13840  | outgoing-2    | sending TCP payload ( 177) FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF 00B1 0104 0D1C 00B4 D5F2 7271 9402 0601 0400 0100 0102 0601 0400 0100 0202 0601 0400 0100 0402 0601 0400 0100 8002 0601 0400 0100 8402 0601 0400 0100 8502 0601 0400 0100 8602 0601 0400 0200 0102 0601 0400 0200 0202 0601 0400 0200 0402 0601 0400 0200 8002 0601 0400 0200 8502 0601 0400 0200 8602 0601 0400 1900 4102 0601 0400 1900 4602 0601 0440 0400 4702 0601 0440 0400 4802 0641 0400 000D 1C02 0206 00
00:05:41 | 13840  | outgoing-2    | >> OPEN version=4 asn=3356 hold_time=180 router_id=213.242.114.113 capabilities=[Multiprotocol(ipv4 unicast,ipv4 multicast,ipv4 nlri-mpls,ipv4 mpls-vpn,ipv4 rtc,ipv4 flow,ipv4 flow-vpn,ipv6 unicast,ipv6 multicast,ipv6 nlri-mpls,ipv6 mpls-vpn,ipv6 flow,ipv6 flow-vpn,l2vpn vpls,l2vpn evpn,bgp-ls bgp-ls,bgp-ls bgp-ls-vpn), Extended Message(65535), ASN4(3356)]
00:05:41 | 13840  | ka-outgoing-2 | receive-timer 60 second(s) left
00:05:41 | 13840  | outgoing-2    | outgoing-2 192.168.88.254-213.242.114.114, closing connection
00:05:41 | 13840  | outgoing-2    | outgoing-2 192.168.88.254-213.242.114.114 213.242.114.114 lost TCP session with peer
00:05:41 | 13840  | outgoing-2    | peer reset, message [closing connection] error[the TCP connection was closed by the remote end]
00:05:41 | 13840  | outgoing-2    | outgoing-2 192.168.88.254-213.242.114.114, closing connection
argument must be an int, or have a fileno() method.

Traceback (most recent call last):
  File "/Users/paunadeurabat/PycharmProjects/RIN/venv/lib/python3.8/site-packages/exabgp/application/control.py", line 245, in run
    self.loop()
  File "/Users/paunadeurabat/PycharmProjects/RIN/venv/lib/python3.8/site-packages/exabgp/application/control.py", line 218, in loop
    ready = self.read_on(reading)
  File "/Users/paunadeurabat/PycharmProjects/RIN/venv/lib/python3.8/site-packages/exabgp/application/control.py", line 104, in read_on
    poller.register(io, select.POLLIN | select.POLLPRI | select.POLLHUP | select.POLLNVAL | select.POLLERR)
TypeError: argument must be an int, or have a fileno() method.
00:05:42 | 13840  | process       | process api-internal-cli-392fd5ca ended, restarting it
00:05:42 | 13840  | process       | terminating process api-internal-cli-392fd5ca
00:05:42 | 13840  | process       | forked process api-internal-cli-392fd5ca
argument must be an int, or have a fileno() method.

Traceback (most recent call last):
  File "/Users/paunadeurabat/PycharmProjects/RIN/venv/lib/python3.8/site-packages/exabgp/application/control.py", line 245, in run
    self.loop()
  File "/Users/paunadeurabat/PycharmProjects/RIN/venv/lib/python3.8/site-packages/exabgp/application/control.py", line 218, in loop
    ready = self.read_on(reading)
  File "/Users/paunadeurabat/PycharmProjects/RIN/venv/lib/python3.8/site-packages/exabgp/application/control.py", line 104, in read_on
    poller.register(io, select.POLLIN | select.POLLPRI | select.POLLHUP | select.POLLNVAL | select.POLLERR)
TypeError: argument must be an int, or have a fileno() method.
00:05:42 | 13840  | process       | process api-internal-cli-392fd5ca ended, restarting it
00:05:42 | 13840  | process       | terminating process api-internal-cli-392fd5ca
00:05:42 | 13840  | process       | forked process api-internal-cli-392fd5ca
argument must be an int, or have a fileno() method.

Traceback (most recent call last):
  File "/Users/paunadeurabat/PycharmProjects/RIN/venv/lib/python3.8/site-packages/exabgp/application/control.py", line 245, in run
    self.loop()
  File "/Users/paunadeurabat/PycharmProjects/RIN/venv/lib/python3.8/site-packages/exabgp/application/control.py", line 218, in loop
    ready = self.read_on(reading)
  File "/Users/paunadeurabat/PycharmProjects/RIN/venv/lib/python3.8/site-packages/exabgp/application/control.py", line 104, in read_on
    poller.register(io, select.POLLIN | select.POLLPRI | select.POLLHUP | select.POLLNVAL | select.POLLERR)
TypeError: argument must be an int, or have a fileno() method.
00:05:42 | 13840  | process       | process api-internal-cli-392fd5ca ended, restarting it
00:05:42 | 13840  | process       | terminating process api-internal-cli-392fd5ca
00:05:42 | 13840  | process       | forked process api-internal-cli-392fd5ca
00:05:42 | 13840  | reactor       | initialising connection to peer-1
00:05:42 | 13840  | outgoing-3    | attempting connection to 213.242.114.114:179
00:05:42 | 13840  | outgoing-3    | sending TCP payload ( 177) FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF 00B1 0104 0D1C 00B4 D5F2 7271 9402 0601 0400 0100 0102 0601 0400 0100 0202 0601 0400 0100 0402 0601 0400 0100 8002 0601 0400 0100 8402 0601 0400 0100 8502 0601 0400 0100 8602 0601 0400 0200 0102 0601 0400 0200 0202 0601 0400 0200 0402 0601 0400 0200 8002 0601 0400 0200 8502 0601 0400 0200 8602 0601 0400 1900 4102 0601 0400 1900 4602 0601 0440 0400 4702 0601 0440 0400 4802 0641 0400 000D 1C02 0206 00
00:05:42 | 13840  | outgoing-3    | >> OPEN version=4 asn=3356 hold_time=180 router_id=213.242.114.113 capabilities=[Multiprotocol(ipv4 unicast,ipv4 multicast,ipv4 nlri-mpls,ipv4 mpls-vpn,ipv4 rtc,ipv4 flow,ipv4 flow-vpn,ipv6 unicast,ipv6 multicast,ipv6 nlri-mpls,ipv6 mpls-vpn,ipv6 flow,ipv6 flow-vpn,l2vpn vpls,l2vpn evpn,bgp-ls bgp-ls,bgp-ls bgp-ls-vpn), Extended Message(65535), ASN4(3356)]
00:05:42 | 13840  | ka-outgoing-3 | receive-timer 60 second(s) left
00:05:42 | 13840  | outgoing-3    | outgoing-3 192.168.88.254-213.242.114.114, closing connection
00:05:42 | 13840  | outgoing-3    | outgoing-3 192.168.88.254-213.242.114.114 213.242.114.114 lost TCP session with peer
00:05:42 | 13840  | outgoing-3    | peer reset, message [closing connection] error[the TCP connection was closed by the remote end]
00:05:42 | 13840  | outgoing-3    | outgoing-3 192.168.88.254-213.242.114.114, closing connection
argument must be an int, or have a fileno() method.

Traceback (most recent call last):
  File "/Users/paunadeurabat/PycharmProjects/RIN/venv/lib/python3.8/site-packages/exabgp/application/control.py", line 245, in run
    self.loop()
  File "/Users/paunadeurabat/PycharmProjects/RIN/venv/lib/python3.8/site-packages/exabgp/application/control.py", line 218, in loop
    ready = self.read_on(reading)
  File "/Users/paunadeurabat/PycharmProjects/RIN/venv/lib/python3.8/site-packages/exabgp/application/control.py", line 104, in read_on
    poller.register(io, select.POLLIN | select.POLLPRI | select.POLLHUP | select.POLLNVAL | select.POLLERR)
TypeError: argument must be an int, or have a fileno() method.
00:05:43 | 13840  | process       | process api-internal-cli-392fd5ca ended, restarting it
00:05:43 | 13840  | process       | terminating process api-internal-cli-392fd5ca
00:05:43 | 13840  | process       | forked process api-internal-cli-392fd5ca
argument must be an int, or have a fileno() method.

Traceback (most recent call last):
  File "/Users/paunadeurabat/PycharmProjects/RIN/venv/lib/python3.8/site-packages/exabgp/application/control.py", line 245, in run
    self.loop()
  File "/Users/paunadeurabat/PycharmProjects/RIN/venv/lib/python3.8/site-packages/exabgp/application/control.py", line 218, in loop
    ready = self.read_on(reading)
  File "/Users/paunadeurabat/PycharmProjects/RIN/venv/lib/python3.8/site-packages/exabgp/application/control.py", line 104, in read_on
    poller.register(io, select.POLLIN | select.POLLPRI | select.POLLHUP | select.POLLNVAL | select.POLLERR)
TypeError: argument must be an int, or have a fileno() method.
00:05:43 | 13840  | process       | process api-internal-cli-392fd5ca ended, restarting it
00:05:43 | 13840  | process       | terminating process api-internal-cli-392fd5ca
00:05:43 | 13840  | process       | forked process api-internal-cli-392fd5ca
00:05:43 | 13840  | process       | Too many death for api-internal-cli-392fd5ca (5) terminating program
00:05:43 | 13840  | reactor       | Problem when sending message(s) to helper program, stopping
00:05:43 | 13840  | reactor       | performing shutdown
00:05:43 | 13840  | process       | terminating process api-internal-cli-392fd5ca
```

```
neighbor 213.242.114.114 {
    router-id 213.242.114.113;
    local-address 192.168.88.254;
    local-as 3356;
    peer-as 33932;
}
```

**Comment by thomas-mangin** [comment]
with your config I get:

```
18:04:13 | 13258  | reactor       | initialising connection to peer-1
18:04:13 | 13258  | outgoing-1    | attempting connection to 213.242.114.114:179
18:04:13 | 13258  | outgoing-1    | connection to 213.242.114.114:179 failed
18:04:13 | 13258  | outgoing-1    | Could not bind to local ip 213.242.114.113 - [Errno 49] Can't assign requested address
18:04:14 | 13258  | outgoing-1    | attempting connection to 213.242.114.114:179
18:04:14 | 13258  | outgoing-1    | connection to 213.242.114.114:179 failed
18:04:14 | 13258  | outgoing-1    | Could not bind to local ip 213.242.114.113 - [Errno 49] Can't assign requested address
18:04:15 | 13258  | outgoing-1    | attempting connection to 213.242.114.114:179
18:04:15 | 13258  | outgoing-1    | connection to 213.242.114.114:179 failed
18:04:15 | 13258  | outgoing-1    | Could not bind to local ip 213.242.114.113 - [Errno 49] Can't assign requested address
```

**Comment by paunadeu** [comment]
What can I do for you? Why I get this error? Can do something for debug deeper?

**Comment by thomas-mangin** [comment]
Nothing you can do really, it is already fixed on master but as I started some work toward what will become 5.0 so the next release will take some time .. I will branch and release a 4.xx release with the fix.

**Comment by thomas-mangin** [comment]
I have released 4.2.7 if you want to see if it fixes the issue.
